src/main/java/com/exasol/adapter/dialects/rewriting/SqlGenerationVisitor.java

```diff
@@ -240,11 +242,31 @@ public String visit(final SqlFunctionAggregateListagg sqlFunctionAggregateListag
     public String visit(final SqlGroupBy groupBy) throws AdapterException {
         final List<String> selectElement = new ArrayList<>();
         for (final SqlNode node : groupBy.getExpressions()) {
-            selectElement.add(node.accept(this));
+            final SqlNode replacement = replaceGroupByExpression(node);
+            selectElement.add(replacement.accept(this));
         }
         return String.join(", ", selectElement);
     }
 
+    /**
+     * Replace an unsupported expression in a {@code GROUP BY} clause with a supported one or return it unchanged.
+     * 
+     * @implNote This replaces numeric literals with their string value as Exasol interprets
+     *           {@code GROUP BY <integer-constant>} as column numbers which was not intended.
+     * 
+     * @param node the original {@code GROUP BY} expression
+     * @return an new, alternative expression or the original expression if no replacement is necessary
+     */
+    private SqlNode replaceGroupByExpression(final SqlNode node) {
```

Comment:
I would rename this method to `workaroundGroupByInteger`. Workarounds should be as visible as possible. Makes it easier to later get rid of them, should the actual source of the problem be fixed.